mdagent/tools/base_tools/analysis_tools/pca_tools.py

```diff
@@ -0,0 +1,370 @@
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import mdtraj as md
+import numpy as np
+import pandas as pd
+import scipy
+import seaborn as sns
+from langchain.tools import BaseTool
+from pydantic import BaseModel, Field
+from sklearn.decomposition import PCA
+
+from mdagent.utils import FileType, PathRegistry
+
+
+class PCA_analysis:
+    def __init__(
+        self,
+        path_registry,
+        pc_percentage,
+        top_path,
+        traj_path,
+        sim_id,
+        selection="backbone",
+    ):
+        self.path_registry = path_registry
+        self.traj = md.load(traj_path, top=top_path)
+        self.sim_id = sim_id
+        if pc_percentage > 1:
+            pc_percentage /= 100
+        self.pc_percentage = pc_percentage
+        self.atom_indices = self.traj.top.select(selection)
+        self.selection = selection
+        self.pc = PCA(n_components=30)
```

Comment:
should n_components be a parameter?

Reply:
maybe, I'm putting 30 here as default, as in some cases no more than 3-4 is necessary to carry >95% of the variance, and otherwise is an indication that the simulation didn't run long enough. 

Im putting 30, and then filtering out to only keep the top 95% of variation. If 30 is not enough... then either the simulation is just random, or has not converged yet (needs more time)

Reply:
But this is up to agreement though, I decided on 30 based on some examples I saw while making the tool

Reply:
hm okay